### PR TITLE
Set up Jest and Enzyme

### PR DESCRIPTION
**Pull request by James-E-Adams**
For some reason it's not recognizing ts/tsx test files, even though this is ported from an existing config which works perfectly.

@Raathigesh any ideas why? @azz might be able to peek in and see if I'm missing anything silly.

Feel free to wait until that's fixed before merging.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -54,15 +55,26 @@
     "typescript-styled-plugin": "^0.2.0"
   },
   "devDependencies": {
+    "@types/jest": "^22.0.1",
     "@types/react": "^16.0.19",
     "@types/react-dom": "^16.0.2",
+    "babel-core": "^6.26.0",
+    "babel-jest": "^22.0.4",
```
Comment (azz) [inline package.json]: Shouldn't need to add this, it's built in to jest.

**Comment by azz** [comment]
You probably want to set [`moduleFileExtensions`](https://facebook.github.io/jest/docs/en/configuration.html#modulefileextensions-array-string) to include `tsx` and `ts`.

File: package.json (hunk 1 above)
Comment (azz) [reply package.json]: Shouldn't actually need anything `babel-*` (including `.babelrc` as there's no JS code in the repo)

Comment (James-E-Adams) [reply package.json]: I added it to get the JS tests working, since TS/JS were both not working.

**Comment by James-E-Adams** [comment]
@azz I tried that in the jest.config.js, and still doesn't work. It's also already in the jest.tsc.config.js.

File: config/jest/common.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+  moduleNameMapper: {
+    '\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$':
+      '<rootDir>/config/jest/fileMock.js',
```
Comment (azz) [inline config/jest/common.js]: These files don't exist.

File: package.json (hunk 1 above)
Comment (azz) [reply package.json]: There's probably a poor caching profile to using both, I'd just go with TS for the tests as well.

Comment (azz) [reply package.json]: Unless you can get `@babel/preset-typesript` working

Comment (James-E-Adams) [reply package.json]: I can't get TS working.

Comment (James-E-Adams) [reply package.json]: Using the config in this PR.

**Comment by azz** [comment]
image

```patch
diff --git a/config/jest/jest.config.js b/config/jest/jest.config.js
index 8a47e83..cf168ed 100644
--- a/config/jest/jest.config.js
+++ b/config/jest/jest.config.js
@@ -4,6 +4,7 @@ module.exports = Object.assign({}, require('./common'), {
   collectCoverage: true,
   mapCoverage: true,
   coverageReporters: ['html'],
+  moduleFileExtensions: ['js', 'ts', 'tsx'],
   testMatch: ['**/*.test.ts', '**/*.test.tsx', '**/*.test.js', '**/*.test.jsx'],
   snapshotSerializers: ['enzyme-to-json/serializer
```

**Comment by Raathigesh** [comment]
I see a screenshot from @azz. Anything I can do to help? 🤔

**Comment by James-E-Adams** [comment]
Damnit  - I was typing `yarn jest` like a fool. Surely you must be jesting, computer gods?!

**Comment by James-E-Adams** [comment]
Thanks @azz, will push somechanges now, and then we should be good to merge @Raathigesh.

**Comment by Raathigesh** [comment]
Thanks @azz

**Comment by azz** [comment]
You can clean up this config significantly: https://github.com/azz/majestic/commit/7553b1827c46ad2e434c3acaaecc7f067f51dbb1

**Comment by Raathigesh** [comment]
@James-E-Adams I suppose the diff lucas posted is the minimum config we need if i'm not mistaken. https://github.com/azz/majestic/commit/7553b1827c46ad2e434c3acaaecc7f067f51dbb1

Can we match that?

**Comment by James-E-Adams** [comment]
@Raathigesh indeed - I'll rebase.

**Comment by James-E-Adams** [comment]
Should be good to go. Shall I give setting up TravisCI a go, so we have tests running on PRs?

**Comment by azz** [comment]
You can probably just copy paste this one ©️ 
https://github.com/Haroenv/dev-ui/blob/master/.travis.yml

**Comment by Raathigesh** [comment]
@James-E-Adams if you could, that would be awesome. Let me know if you need anything from me.

**Comment by James-E-Adams** [comment]
@azz Hah - I was looking at the prettier one. @Raathigesh All good! Will get to it shortly.

**Comment by azz** [comment]
@Raathigesh You'll need to flip the switch on https://travis-ci.org/Raathigesh/majestic.
@James-E-Adams Only thing you need to change is delete the `yarn lint` line.

**Comment by Raathigesh** [comment]
Done!

**Comment by azz** [comment]
Oh, and Electron is a controlled environment so you only need to run tests on one version of Node.js ([8?](https://github.com/electron/electron/blob/master/.node-version))

**Comment by James-E-Adams** [comment]
https://github.com/electron/electron/blob/master/.node-version

we're on 1.7.9, pretty sure Node.js v8 will still be fine.

**Pull request merged**